### PR TITLE
Pass -H:ImageBuildStatisticsFile only when using GraalVM 22.3

### PR DESCRIPTION
**Pull request by zakkak**
Starting with GraalVM 23.0 this option is no longer available. Additionally, `-H:+CollectImageBuildStatistics` is no longer required to get build time information (which was the reason we used it), as that information is now included in the build output json file.

cc @jerboaa, this breaks https://github.com/graalvm/mandrel/pull/405

### CODE REVIEW

**Review by gsmet — APPROVED** [review]

**Comment by zakkak** [comment]
@gsmet please allow @jerboaa to have a look before merging this as it breaks some scripts of his (see https://github.com/graalvm/mandrel/pull/405)

Also note that he is out of office today, hopefully tomorrow he will be able to have a look.

**Review by gsmet — CHANGES_REQUESTED** [review]
Waiting for @jerboaa 's blessing.

**Review by gastaldi — APPROVED** [review]

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
Comment (jerboaa) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java:742]: The way the collector works, is that it collects those metrics files across all integration tests. Thus, it relies on a unique name of the stats files. In this case the native image name was chosen. So please keep that if at all possible. Since https://github.com/oracle/graal/pull/5336 the build stats json file also includes the total time in seconds, which was the main motivating factor for using `-H:+CollectImageBuildStatistics -H:ImageBuildStatisticsFile=...` so that we have a way of knowing the total build time.

Therefore I'd propose a change like so:

```
if (graalVMVersion.compareTo(GraalVM.Version.VERSION_22_3_0) >= 0) {
                    // For getting the build output stats as a JSON file
                    nativeImageArgs.add("-H:BuildOutputJSONFile=" + nativeImageName + "-build-output-stats.json");
}
if (graalVMVersion.compareTo(GraalVM.Version.VERSION_22_3_0) >= 0 && graalVMVersion.compareTo(GraalVM.Version.VERSION_23_0_0) < 0) {
                    // For build time information as 22.3 didn't have total build time in the build stats json
                    nativeImageArgs.add("-H:+CollectImageBuildStatistics");
                    nativeImageArgs.add("-H:ImageBuildStatisticsFile=" + nativeImageName + "-timing-stats.json");
}
```

**Review by jerboaa — CHANGES_REQUESTED** [review]
Thanks for the ping. My suggestions are below. I'll follow up with the relevant updates on the collector side.

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java]: Done

**Review by jerboaa — APPROVED** [review]
LGTM, thanks!

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
Comment (jerboaa) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java]: Thanks!

**Review by gsmet — APPROVED** [review]

**Pull request merged**
